httpbp/client_middlewares.go

```diff
@@ -349,3 +355,26 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 		})
 	}
 }
+
+// ClientHeaderBPMiddleware is a middleware that forwards baseplate headers from the context to the outgoing request.
+//
+// If it detects any new baseplate headers set on the request, it will reject the request and return an error.
+func ClientHeaderBPMiddleware(client string) ClientMiddleware {
```

Comment:
```suggestion
func ClientHeaderBaseplateMiddleware(client string) ClientMiddleware {
```
:no_bell: maybe best spelled out since we have `Baseplate` more often in function or method names than `BP`.

Reply:
`headerbp` is the name of the actual library so I figured it was better to keep it consistent with that?